### PR TITLE
COM-5587 Capture alternate pickup contact

### PR DESCRIPTION
**Pull request by kibo-swapnilpakolu**
COM-5587 added labels in de-DE and deleted duplicate label from en-US

COM-5587 Capture alternate pickup contact

COM-5587 added UI cancel btn and confirmation page

COM-5587 Capture alternate pickup contact

### CODE REVIEW

File: scripts/modules/checkout/models-checkout-page.js [hunk 1]
```diff
@@ -225,6 +225,66 @@ var CheckoutGrouping = Backbone.MozuModel.extend({
...
+                }
+            }
+        },
+        'emailAddress': {
+            fn: function(value,attr) {
+                if(!value) {
+                    return Hypr.getLabel('emailMissing');
+                }
+
+                var emailPattern= /^((([a-z]|\d|[!#\$%&'\*\+\-\/=\?\^_`{\|}~]|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])+(\.([a-z]|\d|[!#\$%&'\*\+\-\/=\?\^_`{\|}~]|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])+)*)|((\x22)((((\x20|\x09)*(\x0d\x0a))?(\x20|\x09)+)?(([\x01-\x08\x0b\x0c\x0e-\x1f\x7f]|\x21|[\x23-\x5b]|[\x5d-\x7e]|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])|(\\([\x01-\x09\x0b\x0c\x0d-\x7f]|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF]))))*(((\x20|\x09)*(\x0d\x0a))?(\x20|\x09)+)?(\x22)))@((([a-z]|\d|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])|(([a-z]|\d|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])([a-z]|\d|-|\.|_|~|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])*([a-z]|\d|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])))\.)+(([a-z]|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])|(([a-z]|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])([a-z]|\d|-|\.|_|~|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])*([a-z]|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])))$/i;
```
Comment (atul-chavan) [inline scripts/modules/checkout/models-checkout-page.js]: Please use regex from already used cases.

File: scripts/pages/multi-ship-checkout.js [hunk 2]
```diff
@@ -169,6 +169,129 @@ require(["modules/jquery-mozu",
...
+            e.preventDefault();
+            this.model.set("firstName",null);
+            this.model.set("lastNameOrSurname",null);
+            this.model.set("emailAddress",null);
+            this.model.set("phoneNumber",null);
+            this.render();
+        },
+        editAlternateContact: function(e) {
+            e.preventDefault();
+            tempAlternateContactData = _.clone(this.model.attributes);//this.model.parent.get("alternateContact");
```
Comment (atul-chavan) [inline scripts/pages/multi-ship-checkout.js]: remove commented code

File: scripts/pages/multi-ship-checkout.js [hunk 3]
```diff
@@ -169,6 +169,129 @@ require(["modules/jquery-mozu",
...
+            this.showForm();
+        },
+        cancelAlternateContact: function(e) {
+            if(tempAlternateContactData) {
+                this.model.attributes = _.clone(tempAlternateContactData);//this.model.parent.set("alternateContact",tempAlternateContactData);
```
Comment (atul-chavan) [inline scripts/pages/multi-ship-checkout.js]: remove commented code

File: scripts/modules/checkout/models-checkout-page.js [hunk 4]
```diff
@@ -876,11 +937,15 @@ var CheckoutPage = Backbone.MozuModel.extend({
                     requiresFulfillmentInfo = this.get('requiresFulfillmentInfo'),
                     requiresBillingInfo = nonStoreCreditOrGiftCardTotal > 0,
                     process = [function() {
-                        return checkout.apiUpdateCheckout({
+                        var processPayload =  {
```
Comment (atul-chavan) [inline scripts/modules/checkout/models-checkout-page.js]: rename it to requestPayload.

File: scripts/pages/multi-ship-checkout.js [hunk 5]
```diff
@@ -169,6 +169,129 @@ require(["modules/jquery-mozu",
...
+        },
+        render: function() {
+            var self = this;
+            Backbone.MozuView.prototype.render.apply(this,arguments);
+                setTimeout(
+                    function() {
+                        var alternatePickupContact= new AlternatePickupView({
+                            el:$('#alternate-contact'),
+                            model: self.model.get("alternateContact")
+                        });
```
Comment (atul-chavan) [inline scripts/pages/multi-ship-checkout.js]: Why duplicate initialization when already done in initialize()?

**Review by atul-chavan — CHANGES_REQUESTED** [review]
Please address changes.

File: scripts/pages/multi-ship-checkout.js (hunk 2 above)
Comment (kibo-swapnilpakolu) [reply scripts/pages/multi-ship-checkout.js]: removed comment

File: scripts/pages/multi-ship-checkout.js (hunk 3 above)
Comment (kibo-swapnilpakolu) [reply scripts/pages/multi-ship-checkout.js]: removed comment

File: scripts/pages/multi-ship-checkout.js (hunk 5 above)
Comment (kibo-swapnilpakolu) [reply scripts/pages/multi-ship-checkout.js]: binding of the click events of child view are getting removed when we call render in parent function. so recreating child view in render of parent view.

File: scripts/modules/checkout/models-checkout-page.js (hunk 1 above)
Comment (kibo-swapnilpakolu) [reply scripts/modules/checkout/models-checkout-page.js]: fixed. used "pattern: email" instead of function.

**Review by atul-chavan — APPROVED** [review]
Looks good to me

**Pull request merged**